elytron/src/main/java/org/wildfly/extension/elytron/CredentialStoreResourceDefinition.java

```diff
@@ -291,6 +294,16 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
             String relativeTo = asStringIfDefined(context, RELATIVE_TO, model);
             ServiceTarget serviceTarget = context.getServiceTarget();
 
+            if (type == null || type.equals(KeyStoreCredentialStore.KEY_STORE_CREDENTIAL_STORE)) {
+                implementationAttributes.putIfAbsent(CS_KEY_STORE_TYPE_ATTRIBUTE, "JCEKS");
+            }
+
+            List<String> filebasedKeystoreTypes = Arrays.asList("JKS", "JCEKS", "PKCS12");
```

Comment:
As this is a constant list it can probably be created once and wrapped in a Collections.unModifiableList() as a static member variable.